### DISCUSSION TITLE
Need to pay $2 every time I connect my farcaster account in the local debugger

### DISCUSSION THREAD

**Discussion by spennyp** [wevm/frog-archived#130] [Ideas]
### Describe the bug

![image](https://github.com/wevm/frog/assets/31705429/301bae5e-73bb-454c-ae79-0b4ba1faa061)

### Steps To Reproduce

1. Start local debugger
2. Connect farcaster with mobile:
![image](https://github.com/wevm/frog/assets/31705429/0fe99840-62ac-4895-88df-e4f3618fbea1)

### Frog Version

0.5.5

### Check existing issues

- [X] I checked there isn't [already an issue](https://github.com/wevm/frog/issues) for the bug I encountered.

**Comment by jaxernst** [comment]
Also ran into this. Docs don't seem to mention anything about this, but apparently these keys are supposed to be stored in cookies? (I accidentally cleared mine)

Would be nice to have some clear docs around this to prevent others from wasting money!

**Comment by blaynem** [comment]
A better way to handle this is by using an ngrok server.

Ngrok is a free tool. Just sign up, then follow the instructions to install it on your machine.

1. Start the `ngrok` server, forwarding your localhost port.
2. Visit the Forwarding Link
    2b. It will open a new browser window, where we click `Visit Site`, which will then reload and you will likely see a big empty screen. THIS IS FINE.
3. Visit `https://warpcast.com/~/developers/frames`, and input the Forwarding link from the step above.

Then you can just login on warpcast itself, rather than attempt to link it.

**Comment by dalechyn** [comment]
I have debugged the devtools and figured out that unless you are passing `hub` to the frog (like neynar), devtools won't be able to rehydrate properly as a request to a hub is made to retrieve user's data such as `displayName` and pfp.

**Comment by dalechyn** [comment]
Will be closed in https://github.com/wevm/frog/pull/306.
